subprojects/scala/src/main/java/org/gradle/api/tasks/ScalaRuntime.java

```diff
@@ -59,9 +62,11 @@
     private static final Pattern SCALA_JAR_PATTERN = Pattern.compile("scala-(\\w.*?)-(\\d.*).jar");
 
     private final Project project;
+    private final JvmEcosystemUtilities jvmEcosystemUtilities;
 
     public ScalaRuntime(Project project) {
         this.project = project;
+        this.jvmEcosystemUtilities = ((ProjectInternal) project).getServices().get(JvmEcosystemUtilities.class);
```

Comment:
Injecting this service via constructor is possible, but this class is in the public API and changing this constructor breaks binary compatibility.